lib/action_subscriber/dsl.rb

```diff
@@ -131,6 +136,41 @@ def _run_action_at_most_once_with_filters(env, action)
       end
     end
 
+    def _run_action_at_most_once_multiple_with_filters(env, action)
+      processed_acknowledgement = false
+      rejected_message = false
+      if env.delivery_tag % multiple_offset == 0 # tags are monotonically increasing integers
```

Comment:
one thing you might run into with keeping a copy of the delivery tag is that march hare will re-use a channel object when it recovers a broken connection, but your delivery tag is no longer valid for that channel.

In other words rabbitmq thinks that this is a new connection and a new channel, but your code is still holding the same channel object. MarchHare does this for convenience to the programmer, but if you try to acknowledge an old delivery tag form the previous version of the channel, it will raise an error.

This error can happen under the current version of `ActionSubscriber`, but batching your acknowledgements just widens the time-period in which an acknowledgement might fail and you have to consider the case that some of your `env`s came from a previous version of the channel.

Reply:
@mmmries isn't that the same as today though? I only added an accessor to read the `delivery_tag` I didn't take a copy of it to store .... basically it will use the same deliver_tag it would have used, it will just use it less often and set the `multiple` header (if this is doing what I think it is)

Reply:
Oh... I see what you are doing here. This basically says to acknowledge all of the messages up to this current delivery tag? Digging through the java it looks like this is reflected at the protocol layer. So we are reducing the total back-and-forths between client and server.

This looks like it should work just fine for `at_most_once` mode.